### PR TITLE
Arsenal - Fix #9040 bugs/regressions

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- Fix script error when saving a loadout with a weapon in a container.
- Readd magazine refill when clicking on equipped magazine in arsenal.

Missed while testing #9040.

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/arsenal/functions/fnc_replaceUniqueItemsLoadout.sqf [hunk 1]
```diff
@@ -98,9 +98,10 @@ private _cfgVehicles = configFile >> "CfgVehicles";
...
                             {
-                                // Magazines
+                                // Magazines in weapons have 2 entries: Name and ammo count
                                 if (_forEachIndex in [4, 5]) then {
-                                    _uniqueBaseCfgText = (getText (_cfgMagazines >> _x >> QGVAR(uniqueBase))) call EFUNC(common,getConfigName);
+                                    _x params ["_magazine"];
+                                    _uniqueBaseCfgText = (getText (_cfgMagazines >> _magazine >> QGVAR(uniqueBase))) call EFUNC(common,getConfigName);
 
                                     if (_uniqueBaseCfgText != "") then {
                                         _weaponsInfo set [_forEachIndex, _uniqueBaseCfgText];
```
Comment (PabstMirror) [inline addons/arsenal/functions/fnc_replaceUniqueItemsLoadout.sqf:107]: is it a problem that we lose the ammo count?

should this be something like 
```
_weaponsInfo set [_forEachIndex, [_uniqueBaseCfgText, _x # 1]];
``

Comment (LinkIsGrim) [reply addons/arsenal/functions/fnc_replaceUniqueItemsLoadout.sqf:107]: It still works and refills the magazine. I don't think it's a problem.

**Review by PabstMirror — APPROVED** [review]

**Comment by johnb432** [comment]
> Readd magazine refill when clicking on equipped magazine in arsenal.

Makes sense for most use cases. However, if the magazine is unique, do we want that?

File: addons/arsenal/functions/fnc_replaceUniqueItemsLoadout.sqf (hunk 1 above)
Comment (johnb432) [reply addons/arsenal/functions/fnc_replaceUniqueItemsLoadout.sqf:107]: Have you tested saving a loadout and reloading it?

I'm have a feeling that this fix can't work, as I had to write [this](https://github.com/acemod/ACE3/blob/93520bdc9d9db63852cfc555595bfc855ae7e52a/addons/arsenal/functions/fnc_replaceUniqueItemsLoadout.sqf#L48) for weapons not in containers. `getUnitLoadout` returns weapons in containers in the same format as weapons not in containers, so I presume the code should be the same and I forgot to correct that in #9040.

I'll take a in-depth look tonight.

**Comment by LinkIsGrim** [comment]
> > Readd magazine refill when clicking on equipped magazine in arsenal.
> 
> Makes sense for most use cases. However, if the magazine is unique, do we want that?

You can already refill unique mags by switching to and from any other magazine anyway. We'd have to keep the magazine's ammo count and restore it if we want consistency.

File: addons/arsenal/functions/fnc_replaceUniqueItemsLoadout.sqf (hunk 1 above)
Comment (johnb432) [reply addons/arsenal/functions/fnc_replaceUniqueItemsLoadout.sqf:107]: It's as I thought unfortunately.

I fixed it in #9297.

**Pull request merged**